cms/wagtail_hooks.py

```diff
@@ -26,3 +30,46 @@ def filter_queryset(self, queryset):
 def configure_admin_api_default_order(router):
     """Swap admin pages API for our own flavor that orders results by title"""
     router.register_endpoint("pages", OrderedPagesAPIEndpoint)
+
+
+@hooks.register("after_publish_page")
+def create_product_and_versions_for_courseware_pages(request, page):
+    """
+    Creates Product and Product Version for the courseware pages on page publish.
+    """
+    if not (
+        page.is_internal_or_external_course_page
+        or page.is_internal_or_external_program_page
+    ):
```

Comment:
This line breaks all the pages except course/program pages because `is_internal_or_external_course_page` is not available in those pages. To reproduce, try publishing any page except course/program page.

Reply:
Great Catch, Migrating from form to hooks caused this issue. Thanks for pointing it out.